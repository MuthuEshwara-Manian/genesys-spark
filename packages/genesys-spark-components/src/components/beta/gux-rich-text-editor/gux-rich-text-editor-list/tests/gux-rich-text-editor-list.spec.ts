import { newSpecPage } from '@test/specTestUtils';

import { GuxRichTextEditorList } from '../gux-rich-text-editor-list';

const components = [GuxRichTextEditorList];
const language = 'en';

describe('gux-rich-text-editor-list', () => {
  it('should build and render', async () => {
    const html = `
    <gux-rich-text-editor-list>
    <gux-rich-style-list-item text-style="heading-1" role="listitem" hydrated="">Heading 1</gux-rich-style-list-item>
    <gux-rich-style-list-item text-style="heading-2" role="listitem" hydrated="">Heading 1</gux-rich-style-list-item>
    <gux-rich-style-list-item text-style="heading-3" role="listitem" hydrated="">Heading 1</gux-rich-style-list-item>
    </gux-rich-text-editor-list>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxRichTextEditorList);
    expect(page.root).toMatchSnapshot();
  });
});
