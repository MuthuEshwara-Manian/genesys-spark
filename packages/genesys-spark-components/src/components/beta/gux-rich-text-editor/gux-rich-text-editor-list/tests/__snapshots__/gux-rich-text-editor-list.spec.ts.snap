// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-rich-text-editor-list should build and render 1`] = `
<gux-rich-text-editor-list role="list">
  <template shadowrootmode="open">
    <span aria-hidden="true" tabindex="-1"></span>
    <slot></slot>
  </template>
  <gux-rich-style-list-item hydrated="" role="listitem" text-style="heading-1">
    Heading 1
  </gux-rich-style-list-item>
  <gux-rich-style-list-item hydrated="" role="listitem" text-style="heading-2">
    Heading 1
  </gux-rich-style-list-item>
  <gux-rich-style-list-item hydrated="" role="listitem" text-style="heading-3">
    Heading 1
  </gux-rich-style-list-item>
</gux-rich-text-editor-list>
`;
